### PR TITLE
Add Photoshop

### PR DESCRIPTION
**Pull request by Kreyren**
Based  on https://appdb.winehq.org/objectManager.php?sClass=version&iId=37541

Depends on: https://github.com/PhoenicisOrg/scripts/issues/909 -> Resolved, Thanks @ImperatorS79 !
Depends on: https://github.com/PhoenicisOrg/scripts/issues/910  -> Resolved, Thanks @ImperatorS79 !
Depends on: https://github.com/PhoenicisOrg/scripts/issues/911  -> Resolved, Thanks @ImperatorS79 !
Depends on: https://github.com/PhoenicisOrg/scripts/issues/912 -> Resolved, Thanks @ImperatorS79 !
Blocked by: https://github.com/PhoenicisOrg/scripts/issues/1012 -> Created https://github.com/PhoenicisOrg/scripts/pull/1060
Blocked by: https://github.com/PhoenicisOrg/scripts/issues/1061 -> Concluded phoenicis-wine issue

Reference: https://www.playonlinux.com/en/app-3489.html

Signed-off-by: Jacob Hrbek <werifgx@gmail.com>

### CODE REVIEW

**Comment by Kreyren** [comment]
Requesting adding multiple versions of photoshop in root of application so that end-user can pick which version they want insted of using `Online`.

**Comment by qparis** [comment]
This is not how we organized the scripts. We should have one different script per version.

**Comment by ImperatorS79** [comment]
Those verbs does not seem very difficult to implement, so @Kreyren I am sure you can do it based another verbs :wink:.

I would however prefer if someone that own Adobe Photoshop could test it under latest wine staging to be sure all those verbs are needed.

**Comment by Zemogiter** [comment]
@Kreyren atmlib, msxml3 and 6 are now part of the repo.

**Comment by Kreyren** [comment]
@Zemogiter Noted, will update once i can

**Comment by ImperatorS79** [comment]
@Kreyren There is now a font_smoothing plugin. Please also check travis and codacy.

**Comment by Kreyren** [comment]
> @Kreyren There is now a font_smoothing plugin. Please also check travis and codacy.

is noted, will do once i have the time

**Comment by Kreyren** [comment]
Results in seemingly java issue https://gist.githubusercontent.com/Kreyren/a5dbad680b9b00bfcd1b188b882ae0b1/raw/63f90df524519070a441b8e567c6b7dd1d3a25f8/dsgadgsdag

help-wanted

**Comment by Kreyren** [comment]
main.png is missing

**Comment by ImperatorS79** [comment]
@Kreyren please test again with those modifications

**Comment by Zemogiter** [comment]
@ImperatorS79 missing `{` after `OnlineInstaller`

**Comment by ImperatorS79** [comment]
How ? In other scripts there is not that.

**Comment by Zemogiter** [comment]
You're right.

File: Applications/Graphics/Photoshop/Online/script.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            .preInstall(function (wine, wizard){
+               wine.corefonts();
+               wine.gdiplus();
+               wine.vcrun2015();
+               wine.atmlib();
+               wine.msxml3();
+               wine.msxml6();
+               wine.amtlib();
+               wine.adobeair();
+               wine.font_smoothing(RGB);
```
[the remark this answers is not in the document]
Comment (ImperatorS79) [reply Applications/Graphics/Photoshop/Online/script.js]: I think it is "RBG"

**Comment by Kreyren** [comment]
QA: It's forcing user-input for MSXML -> Silent install is preffered.

image

**Comment by Kreyren** [comment]
image

Checksum changed

**Comment by madoar** [comment]
The script looks a bit strange to me. I mean you use an `OnlineInstallerScript` but don't use its core functionality, i.e. `.url("http://url_to_exe")`. 

@plata @qparis do you know whether the `OnlineInstallerScript` or `ZipScript` is able to comfortably do the following:
1) download a zip file
2) extract the zip file
3) execute an installer exe file inside the extracted zip file
4) create a shortcut to an exe file created by the installer file

**Comment by Zemogiter** [comment]
@madoar ZipScript

**Comment by madoar** [comment]
@Zemogiter https://phoenicisorg.github.io/scripts/Develop/script-js/ doesn't provide an example how this can be done. Can you add one?

**Comment by Zemogiter** [comment]
I could but I dont remember how to access the `.Phoenicis/resources` path where the file in `url` will be downloaded.

**Comment by plata** [comment]
The problem with this use case is that you must know the filename of the executable inside the script. As this cannot be determined automatically, I don't see an easy way to have the functionality in the QuickScripts. The `ZipScript` will extract the zip content in the container. You could of course use that in `postInstall` but I don't think you want the zip content there. So, unless you think this is a common use case, I would suggest to use a `PlainInstaller`.

**Comment by madoar** [comment]
It seems to me like a more common use-case. I guess a lot of applications try to compress their installers by first archiving it.

**Comment by plata** [comment]
We could enhance the `OnlineInstallerScript` and `LocalInstallerScript`. Maybe an optional parameter:
```Javascript
.executableInArchive("xyz.exe")
```

**Comment by madoar** [comment]
That sounds like a good idea.

**Comment by plata** [comment]
My only problem with this solution is that it increases the complexity of the QuickScript. Also it's not quite clear why it's in `OnlineInstallerScript` and not `ZipScript` (that would be possible as well).

**Comment by Zemogiter** [comment]
I'd say let's do this mod in `ZipScript`

**Comment by plata** [comment]
Ok. Basically adding a `wine.run()` after https://github.com/PhoenicisOrg/scripts/blob/749c609a32aace5376853af643b7d8ebbe012921/Engines/Wine/QuickScript/Zip%20Script/script.js#L56 should be enough.

**Comment by ImperatorS79** [comment]
@Kreyren Could you upload this file as the miniature ?
main

**Comment by ImperatorS79** [comment]
You can also test it by using #1121 and #1074 with this.

**Comment by Kreyren** [comment]
> @Kreyren Could you upload this file as the miniature ?
> main

I can't for legal reasons, sorry..

Note that i no longer contribute to phoenicis, but i keep the repositories up for others to remake the MR or finish it.

**Comment by ImperatorS79** [comment]
The image comes from wikipedia ans is stated as "public domain"...

**Comment by Kreyren** [comment]
> The image comes from wikipedia ans is stated as "public domain"...

Referencing https://www.adobe.com/legal/permissions/trademarks.html

I wasn't able to confirm that said logo is released on wikipedia as public domain, but i nanycase said trademark is still in affect. This usage would be considered as fair use, but i'm not going to merge it on my name.

**Comment by ImperatorS79** [comment]
Okay then ^^.

**Comment by Kreyren** [comment]
> Okay then ^^.

sorry >.>

**Pull request closed without merging**